spring-rabbit/src/main/java/org/springframework/amqp/rabbit/annotation/RabbitListenerAnnotationBeanPostProcessor.java

```diff
@@ -545,7 +545,7 @@ else if (exchangeType.equals(ExchangeTypes.HEADERS)) {
 	private Map<String, Object> resolveArguments(Argument[] arguments) {
 		Map<String, Object> map = new HashMap<String, Object>();
 		for (Argument arg : arguments) {
-			String key = resolveExpressionAsString(arg.name(), "name");
+			String key = resolveExpressionAsString(arg.name(), "@Annotation.name");
```

Comment:
Why `@Annotation`, but not `@Argument` directly?
We have context here, so why just don't reflect it?